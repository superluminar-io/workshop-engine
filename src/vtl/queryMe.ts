export const request = `
  {
    "version": "2018-05-29",
    "payload": {
      "emailAddress": "$ctx.identity.resolverContext.emailAddress",
      "role": "$ctx.identity.resolverContext.role"
    }
  }
`;

export const response = `
  $util.toJson($context.result)
`;
