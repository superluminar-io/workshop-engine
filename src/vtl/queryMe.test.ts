import Parser from 'appsync-template-tester';
import { request, response } from './queryMe';

const me = {
  emailAddress: 'me@example.com',
  role: 'ADMIN',
};

describe('request mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(request);

    const context = {
      identity: {
        resolverContext: me,
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(response);

    const context = {
      result: me,
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
