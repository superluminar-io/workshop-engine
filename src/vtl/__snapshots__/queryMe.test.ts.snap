// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping 1`] = `
Object {
  "payload": Object {
    "emailAddress": "me@example.com",
    "role": "ADMIN",
  },
  "version": "2018-05-29",
}
`;

exports[`response mapping 1`] = `
Object {
  "emailAddress": "me@example.com",
  "role": "ADMIN",
}
`;
